src/classes/CW_AudienceCSVUploadControllerTest.cls

```diff
@@ -0,0 +1,397 @@
+/**
+* File Name   :	CW_AudienceCSVUploadControllerTest.cls
+* Description :	tests CW_AudienceCSVUploadController.cls, CSVIterable.cls, CSVBatchReader,cls
+*
+* Modification Log
+* ============================================================================
+* Ver Date	   Author		 Modification
+* --- ---------- -------------- --------------------------
+* 0.1 02/04/2014 Klea Kolaric	Initial version
+* 0.2 15/04/2014 Klea Kolaric	Changes due to Audience Group Schedule
+*
+*
+*/
+@isTest
+private class CW_AudienceCSVUploadControllerTest 
+{
+	static testMethod void quickScheduleTest()
+	{
+		// test that closest schedule is picked up by default
+		
+		Account acc = TestDataUtil.createPersonAccount();
+		insert acc;
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+		insert camp;
+		
+		// insert default schedule
+		CW_Schedule__c sch1 = TestDataUtil.createSchedule(camp);
+		insert sch1;
+		// insert later schedule
+		CW_Schedule__c sch2 = TestDataUtil.createSchedule(camp);
+		sch2.Start_Date__c = sch1.Start_Date__c.addDays(+2);
+		sch2.End_Date__c = sch1.End_Date__c.addDays(-2);
+		sch2.Evaluation_Start_Date__c = sch2.Start_Date__c;
+		sch2.Evaluation_End_Date__c = sch2.End_Date__c;
+		insert sch2;
+		
+		// insert group
+		CW_Audience_Group__c audGr1 = TestDataUtil.createAudienceGroup(camp.Id);
+		audGr1.Name = group1;
+		audGr1.Is_Control_Group__c = false;
+		insert audGr1;
+		
+		CW_Audience_Group_Schedule__c ags1 = [Select Id from CW_Audience_Group_Schedule__c where Audience_Group__c =: audGr1.id and Schedule__c =:sch1.Id];
+		
+		Test.startTest();
+		CW_Audience_Group_Member__c agm = new CW_Audience_Group_Member__c();
+		agm.Audience_Group_LU__c = audGr1.Id;
+		agm.Contact__c = [select id from Contact where AccountId = :acc.Id limit 1].Id;
+		agm.Audience_Group_Schedule__c = ags1.Id;
+		insert agm;
+		Test.stopTest();
+		
+		System.assertEquals(1,[select count() from CW_Audience_Group_Member__c where id = :agm.Id and Audience_Group_Schedule__c = :ags1.Id]);
+	}
+
+	static testMethod void parseGroupsAndMembers()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility